tidy3d/plugins/mode/mode_solver.py

```diff
@@ -96,23 +103,34 @@ def solver_symmetry(self) -> Tuple[Symmetry, Symmetry]:
 
     @cached_property
     def _solver_grid(self) -> Grid:
-        """Grid for the mode solver, including extension in the normal direction, which is needed
-        to get epsilon from the simulation. The mode fields coordinate along the normal direction
-        will be reset to the exact plane position after the solve."""
+        """Grid for the mode solver, not snapped to plane or simulation zero dims, and also with
+        a small correction for symmetries. We don't do the snapping yet because 0-sized cells are
+        currently confusing to the subpixel averaging. The final data coordinates along the
+        plane normal dimension and dimensions where the simulation domain is 2D will be correctly
+        set after the solve."""
         plane_sym = self.simulation.min_sym_box(self.plane)
-        boundaries = self.simulation.discretize(plane_sym, extend=True).boundaries.to_list
+        monitor = self.to_mode_solver_monitor(name=MODE_MONITOR_NAME)
+        monitor = monitor.updated_copy(colocate=False, center=plane_sym.center, size=plane_sym.size)
+        # pylint:disable=protected-access
+        span_inds = self.simulation._discretize_inds_monitor(monitor)
+
+        # Remove extension along monitor normal
```

Comment:
rather than applying and then removing extensions, would it make sense to parameterize the discretization method to apply or not apply certain extensions?

Reply:
Hmmm. Maybe. Currently it is basically driven by the FieldMonitor-s handling. Maybe it's a matter of taste, but I don't like adding kwargs that do some very specific handling that's only used in one other place. But I could think of something like `extend_zero_dims` which can take care of the first two "fixes". And maybe I could also remove the `plane_sym` thing. It should be possible to just discretize the full plane and then take a subset of indexes if symmetries present.

Reply:
Yeah it's a very minor and local thing, so maybe not worth the time

Reply:
I ended up completely removing `Simulation.min_sym_box` and reorganizing the related symmetry handling. I had my eye on eliminating that some time ago anyway. But yeah apart from that I left the rest as is.